### PR TITLE
chore: use prettier with json

### PR DESCRIPTION
**Pull request by ssbarnea**

### CODE REVIEW

**Review by cidrblock — APPROVED** [review]
More automatic cleanup, ty

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -71,7 +72,12 @@ repos:
     rev: "v2.5.1"
     hooks:
       - id: prettier
-        types: ["markdown"]
+        types_or: ["markdown", "json"]
```
Comment (webknjaz) [inline .pre-commit-config.yaml:75]: It's nicer to have this diff-friendly:
```suggestion
        types_or:
        - markdown
        - json
```

File: .vscodeignore [hunk 2]
```diff
@@ -17,3 +17,4 @@ out/**
 !out/client/extension.js
 server/**
 !out/server/src/server.js
+.pytest_cache
```
Comment (webknjaz) [inline .vscodeignore:20]: This doesn't seem related, submit separately.

File: .ncurc.json [hunk 3]
```diff
@@ -1,6 +1,4 @@
 {
-    "upgrade": true,
-    "reject": [
-      "@types/vscode"
-    ]
+  "upgrade": true,
+  "reject": ["@types/vscode"]
```
Comment (webknjaz) [inline .ncurc.json:3]: Any way not to inline lists?

Comment (ssbarnea) [reply .ncurc.json:3]: No idea as I did not look into it. I see no problem having short lists inline and I am sure that once it will grow enough it will reformat it on multiple lines.

File: .pre-commit-config.yaml (hunk 1 above)
Comment (ssbarnea) [reply .pre-commit-config.yaml:75]: ditto, if formatter didn't do it, i don't care about it.

Comment (webknjaz) [reply .pre-commit-config.yaml:76]: Well, I do — lists are naturally multi-line entries.

File: .ncurc.json (hunk 3 above)
Comment (webknjaz) [reply .ncurc.json:3]: As I said above, lists are naturally multiline things, it's not like we need a compressed JSON. It's not resolved. Don't mark it as resolved arbitrarily. I think it's important to consistently have lists show up as lists.

**Review by priyamsahoo — APPROVED** [review]
